### PR TITLE
Add osdctl command to remediate OCPBUGS-23174

### PR DESCRIPTION
**Pull request by mjlshen**
Hopefully temporary command to help SRE's unstick stuck uninstalling ROSA HCP clusters. This is the only known bug currently with uninstalls currently. Added unit tests/examples for how to run it against one cluster or loop over a bunch of clusters. I expect running against one cluster will be the most common use-case.

That looks like this:

```bash
osdctl cluster cleanup-leaked-ec2 --cluster-id ${CLUSTER_ID}
2024/02/11 16:17:43 searching OCM for cluster: ${CLUSTER_ID}
2024/02/11 16:17:44 cluster ${CLUSTER_ID} found from OCM: ${CLUSTER_ID}
2024/02/11 16:17:44 getting AWS credentials from backplane-api
2024/02/11 16:17:58 context.Background retrieved AWS credentials from backplane-api
2024/02/11 16:18:02 expected instances: map[]
2024/02/11 16:18:03 terminating 1 leaked instances: [i-02906ffde70f270ba]
Continue? (y/N): y
2024/02/11 16:18:06 success - the cluster should be uninstalled soon
```

Check the ticket for output of this looping over stuck uninstalling HCP clusters in prod or a sample stage cluster if you're looking for real output.
 
[OSD-20879](https://issues.redhat.com//browse/OSD-20879)

### CODE REVIEW

File: cmd/cluster/cleanup_leaked_ec2.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    osdctl cluster cleanup-leaked-ec2 --cluster-id ${cluster}
+  done
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.Run(context.Background())
+		},
+	}
+
+	cleanupCmd.Flags().StringVarP(&c.ClusterId, "cluster-id", "C", "", "OCM internal/external cluster id to check for impact of OCPBUGS-23174.")
+	cleanupCmd.Flags().BoolVarP(&c.Yes, "yes", "y", false, "(optional) Skip confirmation prompt when terminating instances")
```
Comment (AlexVulaj) [inline cmd/cluster/cleanup_leaked_ec2.go:61]: Could we use "-S"/"--skip" or something similar for this flag? I think that's what we use elsewhere throughout osdctl and it'd be nice to have consistency.

Comment (mjlshen) [reply cmd/cluster/cleanup_leaked_ec2.go:61]: Oh `-S` is a global flag https://github.com/openshift/osdctl/blob/845f1e8d38660ace6cc2c45d48fe57d1307af36a/internal/utils/globalflags/globalflags.go#L20, so I didn't want to override it here (not sure if it's possible either, haven't tried)

but `-y` is used elsewhere in this repo, like with sending SL's https://github.com/openshift/osdctl/blob/845f1e8d38660ace6cc2c45d48fe57d1307af36a/cmd/servicelog/post.go#L84

**Pull request merged**